library/src/main/java/com/novoda/downloadmanager/VersionOneToVersionTwoMigrator.java

```diff
@@ -0,0 +1,137 @@
+                inputStream = new FileInputStream(new File(originalFileLocation));
+                byte[] bytes = new byte[BUFFER_SIZE];
+
+                // read the v1 file
+                int readLast = 0;
+                while (readLast != -1) {
+                    readLast = inputStream.read(bytes);
+                    if (readLast != 0 && readLast != -1) {
+                        // write the v1 file to the v2 location
+                        internalFilePersistence.write(bytes, 0, readLast);
+                        bytes = new byte[BUFFER_SIZE];
+                    }
+                }
+            } catch (IOException e) {
+                Log.e(getClass().getSimpleName(), e.getMessage());
+                e.printStackTrace();
+            } finally {
+                try {
+                    internalFilePersistence.close();
+                    if (inputStream != null) {
+                        inputStream.close();
+                    }
+                } catch (IOException e) {
+                    Log.e(getClass().getSimpleName(), e.getMessage());
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    private void migrateV1DataToV2Database(List<Migration> migrations) {
+        for (Migration migration : migrations) {
+            migrateV1DataToV2Database(migration);
+        }
+    }
+
+    private void migrateV1DataToV2Database(Migration migration) {
+        Batch batch = migration.batch();
+        downloadsPersistence.startTransaction();
+
+        DownloadBatchTitle downloadBatchTitle = new LiteDownloadBatchTitle(batch.getTitle());
+        DownloadsBatchPersisted persistedBatch = new LiteDownloadsBatchPersisted(downloadBatchTitle, batch.getDownloadBatchId(), DownloadBatchStatus.Status.DOWNLOADED);
+        downloadsPersistence.persistBatch(persistedBatch);
+
+        for (int i = 0; i < migration.originalFileLocations().size(); i++) {
+            String url = batch.getFileUrls().get(i);
+            FileName fileName = LiteFileName.from(batch, url);
+            FilePath filePath = FilePathCreator.create(fileName.name());
+            DownloadFileId downloadFileId = DownloadFileId.from(batch);
+            DownloadsFilePersisted persistedFile = new LiteDownloadsFilePersisted(
+                    batch.getDownloadBatchId(),
+                    downloadFileId,
+                    fileName,
+                    filePath,
+                    migration.fileSizes().get(i).totalSize(),
+                    url,
+                    FilePersistenceType.INTERNAL
+            );
+            downloadsPersistence.persistFile(persistedFile);
+        }
+
+        downloadsPersistence.transactionSuccess();
+        downloadsPersistence.endTransaction();
+    }
+
+    private void deleteFrom(DatabaseWrapper database, List<Migration> migrations) {
+        for (Migration migration : migrations) {
+            String query = "DELETE FROM batches WHERE _id = ?";
```

Comment:
This could be a constant

Reply:
5d0a8bb8a993077824d74d0ac47c851463a43559